sdks/python/apache_beam/yaml/yaml_io_test.py

```diff
@@ -167,6 +167,46 @@ def test_read_with_id_attribute(self):
             result,
             equal_to([beam.Row(payload=b'msg1'), beam.Row(payload=b'msg2')]))
 
+  def test_read_json(self):
+    with beam.Pipeline(options=beam.options.pipeline_options.PipelineOptions(
+        pickle_library='cloudpickle')) as p:
+      with mock.patch('apache_beam.io.ReadFromPubSub',
+                      FakeReadFromPubSub(
+                          topic='my_topic',
+                          messages=[PubsubMessage(
+                              b'{"generator": {"x": 0, "y": 0}, "rank": 1}',
+                              {'weierstrass': 'y^2+y=x^3-x', 'label': '37a'})
+                                    ])):
+        result = p | YamlTransform(
```

Comment:
can we have some error case tests here to improve the coverage?

Reply:
Done. This actually uncovered a bug when transforms are explicitly named in some cases. Thanks.